src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1610,8 +1613,10 @@ class ApplicationManagerImpl
     mobile_apis::SystemContext::eType system_context;
   };
 
-  hmi_apis::HMI_API* hmi_so_factory_;
-  mobile_apis::MOBILE_API* mobile_so_factory_;
+  std::unique_ptr<hmi_apis::HMI_API> hmi_so_factory_;
+  std::unique_ptr<mobile_apis::MOBILE_API> mobile_so_factory_;
+  std::unique_ptr<ns_smart_device_link_rpc::V1::v4_protocol_v1_2_no_extra>
+      v4_protocol_so_factory_;
```

Comment:
@ychernysheva as long as you are returning factories by reference, there is no sense to keep them in unique ptrs. Suggest just to make as a class members:
```
hmi_apis::HMI_API hmi_so_factory_;
mobile_apis::MOBILE_API mobile_so_factory_;
ns_smart_device_link_rpc::V1::v4_protocol_v1_2_no_extra v4_protocol_so_factory_;
```
Then you don't need to make a trick with get() + dereference

Reply:
@AKalinich-Luxoft Done in c5bd07e